main.go

```diff
@@ -120,3 +134,37 @@ func validateDatabaseFlags(fatal bool, path bool, host bool, username bool, pass
 		glog.Fatal("Invalid flag(s) were passed, are you using the correct volume driver?")
 	}
 }
+
+func getStorageConnection() (drivers.StorageController, error) {
+	glog.Info("Attempting to use the ", storageControllerDriver, " storage controller.")
+
+	switch storageControllerDriver {
+	case "on-disk":
+		validateStorageFlags(true, true)
+		return drivers.NewOnDiskStorageController(storageControllerPath), nil
+
+	case "glusterfs":
+		validateStorageFlags(false, false)
+		return drivers.NewGlusterStorageController(), nil
+
+	default:
+		return nil, errors.New("Unsupported storage controller, please choose glusterfs or on-disk.")
+	}
+}
+
+func validateStorageFlags(fatal bool, path bool) {
+
+	var errors bool
+	noteErrorFunc := func(name string, value string, used bool) {
```

Comment:
Since we only invoke this anonymous function once, we should probably just write the logic in-line